### ISSUE TITLE
Problems making Certificate Request on most recent commit. 

### ISSUE THREAD

**Issue by jcll** [mail-in-a-box/mailinabox#1213]
Issue with initial SSL certificate on the past few commits.  Was working on previous direct pulled builds and works on the most recent release. Sorry that I am unable to determine when the breakage occurred.   

```
Archiving old resolv.conf (was /etc/resolvconf/resolv.conf.d/original, now /etc/resolvconf/resolv.conf.original).
Creating initial SSL certificate and perfect forward secrecy Diffie-Hellman parameters...

FAILED: openssl req -new -key /home/user-data/ssl/ssl_private_key.pem -out /tmp/ssl_cert_sign_req-4983.csr -sha256 -subj /C=/ST=/L=/O=/CN=-Redacted-
-----------------------------------------
problems making Certificate Request
140657451689792:error:0D07A098:asn1 encoding routines:ASN1_mbstring_ncopy:string too short:../crypto/asn1/a_mbstr.c:102:minsize=2

```

**Comment by yodax** [comment]
I can confirm this. I don't think it's because of recent commits, when I go back several releases it still fails. 

The problem is that suddenly the country, state, location and organisation fields are enforced. I don't know why. (A quick search didn't yield any results) When I change the line to this it works:

```
openssl req -new -key /home/user-data/ssl/ssl_private_key.pem -out /tmp/ssl_cert_sign_req-22894.csr -sha256 -subj /C=AA/ST=AA/L=Internet/O=MailInABox/CN=box.scw-1a66fd.space
```

This is the openssl version:

```
openssl version
OpenSSL 1.1.0f  25 May 2017
```

Fairly recent, but I don't think the problems started on the 25th of may. We would've heard something. I can't find anything in the openssl changelog about this either.

A fix for this would be changing the way we create a CSR to what I wrote above. But we also do it [here](https://github.com/mail-in-a-box/mailinabox/blob/master/management/ssl_certificates.py#L559). I would like to know what changed.

Renewing certificates with let's encrypt still works fine.

**Comment by teaborn** [comment]
I had the same problem.

This problem comes from the new repository "http://ppa.launchpad.net/ondrej/php/ubuntu/" for php7. There is also a newer "openssl" package which isn't compatible with the system.

In a fresh installation you have to modify the ssl.sh setup:
sudo sed -i -e "/^apt_install/s/^/#/" setup/ssl.sh

For removing the bad package and marking it to not update, you have to enter:
sudo apt-get install openssl=1.0.1f-1ubuntu2.22
sudo apt-mark hold openssl

**Comment by yodax** [comment]
Thanks for figuring that out. That also explains why me revering back to an old version after the PPA was added didn't have any effect.

Since this is just used to create a self signed certificate, which is replaced by a lets encrypt one right after we could just change the CSR and make it work. 

Am I correct in saying that if we hold the openssl package it won't be automatically updated?

**Comment by teaborn** [comment]
Thats correct and this a problem, because on security updates from the openssl package we didn't get the last recent version. I didn't found a solution to exclude a package from a specific repository. The most easy solution is to take the php7 package in the mail-in-a-box repository, but than you have to maintain the security updates on your own for the php7 package.

**Comment by yodax** [comment]
I found a way to not pin it to a specific version but to a specific source. This should allow automatic updates to work. See the [#1221](https://github.com/mail-in-a-box/mailinabox/pull/1221) for extra info. If you could test this it would be appreciated.

**Comment by JoshData** [comment]
Fixed by #1223 hopefully.
